stdlib/Profile/src/Profile.jl

```diff
@@ -220,34 +246,35 @@ function print(io::IO,
                     any_nosamples = true
                 else
                     nl = length(taskids) > 1 ? "\n" : ""
-                    printstyled(io, "Thread $threadid$nl"; bold=true, color=Base.info_color())
+                    printstyled(ioc, "Thread $threadid$nl"; bold=true, color=Base.info_color())
                     for taskid in taskids
-                        printstyled(io, " Task $(Base.repr(taskid))\n"; bold=true, color=Base.debug_color())
-                        nosamples = print(io, data, lidict, pf, format, threadid, taskid, true)
+                        printstyled(ioc, " Task $(Base.repr(taskid)) "; bold=true, color=Base.debug_color())
+                        nosamples = print(ioc, data, lidict, pf, format, threadid, taskid, true)
                         nosamples && (any_nosamples = true)
-                        println(io)
+                        println(ioc)
                     end
                 end
             end
         elseif groupby == :task
             threads = 1:typemax(Int)
             for taskid in intersect(get_task_ids(data), tasks)
-                printstyled(io, "Task $(Base.repr(taskid))\n"; bold=true, color=Base.debug_color())
-                nosamples = print(io, data, lidict, pf, format, threads, taskid, true)
+                printstyled(ioc, "Task $(Base.repr(taskid)) "; bold=true, color=Base.debug_color())
+                nosamples = print(ioc, data, lidict, pf, format, threads, taskid, true)
                 nosamples && (any_nosamples = true)
-                println(io)
+                println(ioc)
             end
         elseif groupby == :thread
             tasks = 1:typemax(UInt)
             for threadid in intersect(get_thread_ids(data), threads)
-                printstyled(io, "Thread $threadid\n"; bold=true, color=Base.info_color())
-                nosamples = print(io, data, lidict, pf, format, threadid, tasks, true)
+                printstyled(ioc, "Thread $threadid "; bold=true, color=Base.info_color())
+                nosamples = print(ioc, data, lidict, pf, format, threadid, tasks, true)
                 nosamples && (any_nosamples = true)
-                println(io)
+                println(ioc)
             end
         end
         any_nosamples && warning_empty(summary = true)
     end
+    Base.print(io, String(resize!(iob.data, iob.size)))
```

Comment:
Why not do this in the `print(::Vector{<:Unsigned})` method so that it's still possible to use `print(::IO, ::Vector{<:Unsigned})` with less allocations? This is only required for printing stuff to `stdout` in one shot, right?

Reply:
Good point! Yeah, this is to enable it to print to stdout in one shot. I was unaware of the work done to avoid stackoverflows during profile prints, so it does make sense to make this optional though.

Reply:
I made this change/revert and concluded that the `print(::Vector{<:Unsigned})` method was already covered by the main one, except for defaulting the io to stdout, so I removed that method and just set the default io in the main method.
Does that sound reasonable?

Reply:
Hmm... it looks like you removed the method `print(::Vector{<:Unsigned})`? Does `Profile.print(Profile.fetch())` work with this PR?

My suggestion was something like:

```julia
function print(data::Vector{<:Unsigned} = fetch(include_meta = true), lidict::Union{LineInfoDict, LineInfoFlatDict} = getdict(data); kwargs...)
    str = sprint(context = stdout) do io
        print(io, data, lidict; kwargs...)
    end
    Base.print(stdout, str)
end
```

(or maybe we need `context = IOContext(stdout, :displaysize=>displaysize(stdout))`?)

Reply:
Yes, sorry, I was getting confused by the positional arg defaults. I've added the method back.
I chose to implement the buffered print in `_peek_report` specifically, because it would seem odd to do it in all cases.